streams/src/main/java/org/apache/kafka/streams/state/internals/TimestampedWindowStoreBuilder.java

```diff
@@ -56,6 +59,11 @@ public TimestampedWindowStoreBuilder(final WindowBytesStoreSupplier storeSupplie
                 store = new InMemoryTimestampedWindowStoreMarker(store);
             }
         }
+        if (storeSupplier.retainDuplicates()) {
```

Comment:
```suggestion
        if (storeSupplier.retainDuplicates() && enableCaching) {
```
Should we only log if we're changing the configured caching? (Also applies below)

Reply:
Good point. Pushed the change